###  importing libraries for loading data and visulization


In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import math
import glob
import cv2 as cv
import pickle
import os
from skimage.io import imread_collection


### loading dataset


In [4]:
# 1
# loading images randomly
# images = [cv.imread(file) for file in glob.glob("../dataset/one/files/*.jpg")]


# 2
# this is loading images randomly
# def load_images_from_folder(folder):
#     images = []
#     for filename in os.listdir(folder):
#         img = cv.imread(os.path.join(folder,filename))
#         if img is not None:
#             images.append(img)
#     return images

# images = load_images_from_folder('../dataset/one/files/')

# 3
# changes the rgb value of image loaded
# #your path 
# col_dir = '../dataset/one/files/*.jpg'

# #creating a collection with the available images
# images = imread_collection(col_dir)



# 4
# this works
filenames = [img for img in glob.glob("../dataset/one/files/*.jpg")]

filenames.sort() # ADD THIS LINE

images = []
for img in filenames:
    n= cv.imread(img)
    images.append(n)


In [5]:
print(type(images))
type(images[1])

<class 'list'>


numpy.ndarray

In [6]:
print(images[100].shape)
images[3].shape


(192, 180, 3)


(192, 179, 3)

In [8]:
cv.imshow('check', images[0])
cv.waitKey(0)
cv.destroyAllWindows()

### some images have 192*180 dimentions whereas some has 192 * 179
### we'll scale each to 192*180

In [9]:
temp  = cv.resize(images[1], (192,180) )
print(type(temp))

<class 'numpy.ndarray'>


In [10]:
images_resized = []
for image in images:
    images_resized.append(cv.resize(image, (180,192) ) )

### now images_resized contains all images of resolution 192*180

In [11]:
print(len(images_resized))
print(images_resized[1].shape)

for index,image in enumerate(images_resized):
    if(image.shape != (192,180,3)):
        print(index)
#nothing means all images scaled properly

4000
(192, 180, 3)


In [12]:
cv.imshow('check', images_resized[1000])
cv.waitKey(0)
cv.destroyAllWindows()

In [14]:
# with open('X_train.txt', 'w') as f:
#     for image in images:
#         f.write("%s\n" % image)

In [15]:
# with open('X_train.txt','r') as f:
#     for image in images

### using pickle to save and load data

In [17]:
#save to X_train
with open('../temp_data_file/X_train', 'wb') as fp:
    pickle.dump(images_resized, fp)


In [ ]:
#load from X_train
with open ('../temp_data_file/X_train', 'rb') as fp:
    itemlist = pickle.load(fp)
    

In [ ]:
cv.imshow('check', itemlist[0])
cv.waitKey(0)
cv.destroyAllWindows()
#working